packages/react-composites/tests/browser/common/utils.ts

```diff
@@ -238,7 +238,17 @@ export const waitForPiPiPToHaveLoaded = async (page: Page, videosEnabledCount: n
 export const stubMessageTimestamps = async (page: Page): Promise<void> => {
   const messageTimestampId: string = dataUiId(IDS.messageTimestamp);
   await page.evaluate((messageTimestampId) => {
-    Array.from(document.querySelectorAll(messageTimestampId)).forEach((i) => (i.innerHTML = 'timestamp'));
+    Array.from(document.querySelectorAll(messageTimestampId)).forEach((i) => (i.textContent = 'timestamp'));
+    (function checkForStubbedTimestamps(i) {
+      setTimeout(function () {
```

Comment:
avoid our own setTimeouts, instead use `waitForFunction`

so something like:
```ts
// Set timestamps in the DOM
await page.evaluate((messageTimestampId) => {
  Array.from(document.querySelectorAll(messageTimestampId)).forEach((i) => (i.textContent = 'timestamp'));
}, messageTimestampId);

// Wait for timestamps to have been updated in the DOM
// eslint-disable-next-line @typescript-eslint/no-explicit-any
await waitForFunction(page, (args: any) => {
  const timestampNodes = document.querySelectorAll(args.messageTimestampId);
  return timestampNodes.every(node => node.innerHTML === 'timestamp');
}, {
  messageTimestampId: messageTimestampId
})
```
(^disclaimer: completely untested code written in the github browser comment window, likely has syntax and logical errors)

Using waitForFunction both allows avoiding custom timeouts and uses the global timeout system, but is also wrapped in a takeScreenshotOnFailure in the event of a timeout happening.

Reply:
waitForFunction will also fail the test here if the timestamps never updated

Reply:
Thanks for the code snippet. It worked for me. Before I commit this, I was wondering does `waitForFunction` retry when `timestampNodes.every(node => node.innerHTML === 'timestamp')` returns false?

Reply:
It should do yep. It should poll every x seconds (or every animationFrameRequest depending on what the default polling type is) and then has an overall timeout: https://playwright.dev/docs/api/class-page#page-wait-for-function (which we limit to 5 seconds)
![image](https://user-images.githubusercontent.com/2684369/162291786-094bc1df-a83f-4e41-9134-6904fff6bd53.png)